### PR TITLE
Add interfaces + Clean weird typing/errors

### PR DESCRIPTION
**Pull request by AndreasPB**
## TypeScript overhaul
* Adds interfaces (Provider, Media, Meilisearch)
* Renames media to meilisearch
* Removes redundant type-hints
* Fixes wrong/incomplete typing
* Fixes being lazy `any` and no type errors

### CODE REVIEW

File: frontend/src/routes/index.svelte [hunk 1]
```diff
@@ -36,10 +37,10 @@
   }
 
   // TODO: Replace any with a Media type
```
Comment (nullxDEADBEEF) [inline frontend/src/routes/index.svelte:39]: I think that TODO is handled now :D

**Review by nullxDEADBEEF — COMMENTED** [review]
I'm not sure what to think of going back to the index page when changing the theme. LGTM otherwise with the very minor change :D

**Comment by AndreasPB** [comment]
Oh, right... I only tested it on the index page... I wanted it to be the same page you're on :D How does one do that?

File: frontend/src/components/navbar.svelte [hunk 2]
```diff
@@ -80,12 +80,12 @@
             on:click={() => ($chosenTheme = theme.value)}
           >
             {#if $chosenTheme == theme.value}
-              <a class="bg-primary hover:bg-primary">
+              <a href="/" class="bg-primary hover:bg-primary">
```
Comment (nullxDEADBEEF) [inline frontend/src/components/navbar.svelte:83]: the href is back :O

**Review by nullxDEADBEEF — CHANGES_REQUESTED** [review]

File: frontend/src/routes/index.svelte [hunk 3]
```diff
@@ -72,8 +72,9 @@
               `&limit=${currentMediaAmount}`
           )
     $inputQuery = input
-    media = await res.json()
-    hitProviderAmounts(media.hits)
+    meilisearch = await res.json()
+    console.log(meilisearch)
```
[the remark this answers is not in the document]
Comment (AndreasPB) [reply frontend/src/routes/index.svelte]: :hear_no_evil:

File: frontend/src/utils.ts [hunk 4]
```diff
@@ -56,7 +56,7 @@ export const removeContentWithMissingImagePath = (list: [], pathName: string) =>
 }
 
 // TODO: Replace any with Media type
-export const sortListByPopularity = (list: [any]) => {
+export const sortListByPopularity = (list: any[]) => {
```
[the remark this answers is not in the document]
Comment (AndreasPB) [reply frontend/src/utils.ts]: yuup, it uses `.popularity`

File: frontend/src/components/navbar.svelte (hunk 2 above)
Comment (AndreasPB) [reply frontend/src/components/navbar.svelte]: delete it, but now we have the warnings back

Comment (nullxDEADBEEF) [reply frontend/src/components/navbar.svelte]: Let's handle the warning in a different issue :DD

**Pull request merged**